### PR TITLE
feat: allow cancelled replace requests to be executed

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

**Review by sander2 — CHANGES_REQUESTED** [review]
Can you add an integration test for executing a cancelled replace where there was a non-zero additional-collateral amount supplied by the new vault? I think that might be not handled correctly with this pr. 

You can use this function with a non-zero amount: https://github.com/interlay/interbtc/blob/master/standalone/runtime/tests/mock/replace_testing_utils.rs#L61

File: crates/replace/src/lib.rs [hunk 1]
```diff
@@ -501,17 +501,25 @@ impl<T: Config> Pallet<T> {
...
+            collateral
         } else if replace.status == ReplaceRequestStatus::Cancelled {
             // we need to re-accept first, this will check that the vault is over the secure threshold
             Self::accept_replace_tokens(&old_vault_id, &new_vault_id, &amount)?;
-        }
+            // no additional collateral locked for this
+            Amount::zero(collateral.currency())
+        } else {
+            // we never enter this branch as completed requests are filtered
+            Amount::zero(collateral.currency())
```
Comment (sander2) [inline crates/replace/src/lib.rs:521]: Instead of the if-elses let's do a pattern match, and return an error when it's not pending/cancelled? I realize that it doesn't really matter since the code path should never be taken, but it's just cleaner

**Review by sander2 — APPROVED** [review]
